sdk/src/process_instruction.rs

```diff
@@ -204,6 +234,21 @@ impl BpfComputeBudget {
             max_cpi_instruction_size: 1280, // IPv6 Min MTU size
             cpi_bytes_per_unit: 250,        // ~50MB at 200,000 units
             sysvar_base_cost: 100,
+            bignum_from_u32_base_cost: 100,
+            bignum_from_dec_str_base_cost: 100,
+            bignum_mod_exp_base_cost: 100,
+            bignum_log_cost: 100,
+            bignum_add_cost: 30,
+            bignum_sub_cost: 30,
+            bignum_mul_cost: 60,
```

Comment:
add and sub are super cheap operations, you can just iterate through the `u64`s and that's it. The mul/div/exp etc way much more expensive. Are these numbers right?

Reply:
Had reviewed with @jackcmay prior, open to changes but I'd also note that there is a bytes cost associated with the `lhs` and `rhs` sizes

Reply:
>     * Solidity has fixed-length signed/unsigned integer type
> 
>     * The type length is a multiple of 8 and between 8 and 256. So you can have `uint8`, `uint72`, `int224`, and `uint256`.
> 
>     * This bignum implementation can handle types which are a multiple of 64 bit, not 8 bit; so we do a `uint112` add we need to know if the result can fit into a `uint112`.
> 
>     * The sign bit is stored in the highest bit for signed types. This bignum stores the sign separately.
> 
> 
> So I don't think this is usuable as-is for solang, I'm afraid.
> 
> I do have some other comments after reviewing the code, I'll add them as a comment to the code.

Looks like I as wrong, this bignum can deal with any type of a multiple of 8 bits. Two issues remain which make it unusable:

- The sign is stored separately rather than as highest bit. This means overflow does not make sense for signed types.
- Overflow causes the BPF program to abort. In Solidity, you can turn off overflow checks in some code blocks. So this needs to be optional.

Reply:
Ok, I missed that. As long as has been given due attention that's fine.

Reply:
cost estimation is still a rough game :-(